project/src/test/java/com/google/sps/OpportunitySignupFormHandlerTest.java

```diff
@@ -0,0 +1,93 @@
+package com.google.sps;
+
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.sps.data.VolunteeringOpportunity;
+import com.google.sps.servlets.OpportunitySignupFormHandlerServlet;
+import com.google.sps.utilities.SpannerClient;
+import com.google.sps.utilities.SpannerTasks;
+import com.google.sps.utilities.SpannerTestTasks;
+import java.io.IOException;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+import org.springframework.mock.web.MockServletContext;
+
+
+/** Test that tests the opportunity signup form functionality. */
+@RunWith(JUnit4.class)
+public final class OpportunitySignupFormHandlerTest {
+  private static final String EVENT_ID = "0883de79-17d7-49a3-a866-dbd5135062a8";
+  private static final String NAME = "Performer";
+  private static final int NUMBER_OF_SPOTS = 240;
+  private static final String OPPORTUNITY_ID = "opportunity-id";
+  private static final String EMAIL = "email";
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalUserServiceTestConfig());
+
+  @Before
+  public void setUp() throws Exception {
+    // Mock a request to trigger the SpannerClient setup to run
+    MockServletContext mockServletContext = new MockServletContext();
+    new SpannerClient().contextInitialized(new ServletContextEvent(mockServletContext));
+    SpannerTestTasks.setup();
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    SpannerTestTasks.cleanup();
+    helper.tearDown();
+  }
+
+  @Test
+  public void testAddOpportunitySignup_LoggedIn() throws IOException {
+    VolunteeringOpportunity opportunity =
```

Comment:
For each line of this part, please make such a decision: If the reader doesn't see this line, can the reader still understand the test.

If the answer is yes, move the line to setUp.

Reply:
Done - thanks.